packages/gatsby-graphiql-explorer/src/app/app.js

```diff
@@ -0,0 +1,159 @@
+import React from "react"
+import ReactDOM from "react-dom"
+
+import GraphiQL from "graphiql"
+import GraphiQLExplorer from "graphiql-explorer"
+import { getIntrospectionQuery, buildClientSchema } from "graphql"
+
+import "whatwg-fetch"
+
+import "graphiql/graphiql.css"
+import "./app.css"
+
+const parameters = {}
+window.location.search
+  .substr(1)
+  .split(`&`)
+  .forEach(function(entry) {
+    var eq = entry.indexOf(`=`)
+    if (eq >= 0) {
+      parameters[decodeURIComponent(entry.slice(0, eq))] = decodeURIComponent(
+        entry.slice(eq + 1)
+      )
+    }
+  })
+// Produce a Location query string from a parameter object.
+function locationQuery(params) {
+  return (
+    `?` +
+    Object.keys(params)
+      .filter(function(key) {
+        return Boolean(params[key])
+      })
+      .map(function(key) {
+        return encodeURIComponent(key) + `=` + encodeURIComponent(params[key])
+      })
+      .join(`&`)
+  )
+}
+
+// Derive a fetch URL from the current URL, sans the GraphQL parameters.
+const graphqlParamNames = {
+  query: true,
+  variables: true,
+  operationName: true,
+}
+const otherParams = {}
+for (var k in parameters) {
+  if (parameters.hasOwnProperty(k) && graphqlParamNames[k] !== true) {
+    otherParams[k] = parameters[k]
+  }
+}
+const fetchURL = locationQuery(otherParams)
+
+function graphQLFetcher(graphQLParams) {
+  return fetch(fetchURL, {
+    method: `post`,
+    headers: {
+      Accept: `application/json`,
+      "Content-Type": `application/json`,
+    },
+    body: JSON.stringify(graphQLParams),
+    credentials: `include`,
+  }).then(function(response) {
+    return response.json()
+  })
+}
+
+// When the query and variables string is edited, update the URL bar so
+// that it can be easily shared.
+function onEditVariables(newVariables) {
+  parameters.variables = newVariables
+  updateURL()
+}
+function onEditOperationName(newOperationName) {
+  parameters.operationName = newOperationName
+  updateURL()
+}
+function updateURL() {
+  history.replaceState(null, null, locationQuery(parameters))
+}
+
+// We control query, so we need to recreate initial query text that show up
+// on visiting graphiql - in order it will be
+//  - query from query string (if set)
+//  - query stored in localStorage (which graphiql set when closing window)
+//  - default empty query
+const DEFAULT_QUERY =
+  parameters.query ||
+  (window.localStorage && window.localStorage.getItem(`graphiql:query`)) ||
+  `{
+
+}`
+
+class App extends React.Component {
+  state = { schema: null, query: DEFAULT_QUERY, explorerIsOpen: true }
```

Comment:
explorer is open by default - probably would be better to track state and save it to localStorage

Reply:
I like the idea of exposing it by default. It's a useful feature, let's make it obvious! 💪

Reply:
I should be more clear - I mean it will always start with explorer opened, so even if user close this and never use this, next time he will visit `/___graphiql` explorer will be opened again. This might be frustrating for users pretty familiar with gatsby schema that won't benefit that much from explorer

Reply:
No - I got that! I was commenting on the former part of the statement. I do think it could make sense to save a key to localStorage once closed (and opened).

Reply:
done